front/lib/api/files/preprocessing.ts

```diff
@@ -168,6 +170,52 @@ const extractTextFromPDF: PreprocessingFunction = async (
   }
 };
 
+// CSV preprocessing.
+// We upload the content of the CSV on the processed bucket and the schema in the snippet bucket.
+const extractContentAndSchemaFromCSV: PreprocessingFunction = async (
+  auth: Authenticator,
+  file: FileResource
+): Promise<Result<undefined, Error>> => {
+  try {
+    const readStreamForProcessedFile = file.getReadStream(auth, "original");
+    const processedWriteStream = file.getWriteStream(auth, "processed");
+    await pipeline(readStreamForProcessedFile, processedWriteStream);
+
+    const readStreamForSnippetFile = file.getReadStream(auth, "original");
+    const schemaWriteStream = file.getWriteStream(auth, "snippet");
+
+    await pipeline(
+      readStreamForSnippetFile,
+      parse({
+        columns: true,
+        skip_empty_lines: true,
+        trim: true,
+      }),
+      analyzeCSVColumns, // Use the generator function to process rows
+      async function* (schema) {
+        for await (const schemaData of schema) {
+          yield JSON.stringify(schemaData, null, 2);
```

Comment:
So the snippet doesn't have the few first lines of the file ?

Reply:
Yes, only the schema!

Reply:
Do we feel this will be more valuable to the model vs having the table head ?

Reply:
Yes, this was the intention.